pkg/steps/release/promote_test.go

```diff
@@ -503,14 +515,62 @@ func TestPromotedTagsWithRequiredImages(t *testing.T) {
 					{Namespace: "ocp", Name: "4.6", Tag: "base-8"},
 				},
 			},
+			expectedNames: sets.New[string]("base", "base-7", "base-8"),
+		},
+		{
+			name: "promotion with multiple to stanzas",
+			input: &api.ReleaseBuildConfiguration{
+				PromotionConfiguration: &api.PromotionConfiguration{
+					Targets: []api.PromotionTarget{{
+						ExcludedImages: []string{"*"},
+						AdditionalImages: map[string]string{
+							"other": "base",
+						},
+						Namespace: "extra",
+						Tag:       "latest",
+					}},
+					Namespace: "ocp",
+					Name:      "4.6",
+					AdditionalImages: map[string]string{
+						"base":   "base-8",
+						"base-7": "base-7",
+						"base-8": "base-8",
+					},
+				},
+				Metadata: api.Metadata{
+					Org:    "openshift",
+					Repo:   "images",
+					Branch: "release-4.6",
+				},
+			},
+			expected: map[string][]api.ImageStreamTagReference{
+				"base": {
+					{Namespace: "extra", Name: "other", Tag: "latest"},
+				},
+				"base-7": {
+					{Namespace: "ocp", Name: "4.6", Tag: "base-7"},
+				},
+				"base-8": {
+					{Namespace: "ocp", Name: "4.6", Tag: "base"},
+					{Namespace: "ocp", Name: "4.6", Tag: "base-8"},
+				},
+			},
+			expectedNames: sets.New[string]("base", "base-7", "base-8", "other"),
 		},
 	}
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			mapping, _ := PromotedTagsWithRequiredImages(testCase.input, testCase.options...)
-			if actual, expected := mapping, testCase.expected; !reflect.DeepEqual(actual, expected) {
-				t.Errorf("%s: got incorrect promoted tags: %v", testCase.name, diff.ObjectDiff(actual, expected))
+			mapping, names := PromotedTagsWithRequiredImages(testCase.input, testCase.options...)
+			actual, expected := mapping, testCase.expected
+			if diff := cmp.Diff(actual, expected); diff != "" {
+				t.Errorf("%s: got incorrect promoted tags: %v", testCase.name, diff)
+			}
+			if !names.Equal(testCase.expectedNames) {
+				actual, expected := names.UnsortedList(), testCase.expectedNames.UnsortedList()
```

Comment:
Do we need to sort them?

Set defines the `Equal` function like

https://github.com/openshift/ci-tools/blob/9a37c4b4c981d5c2ab97bfe41aebc9aa07fc2178/vendor/k8s.io/apimachinery/pkg/util/sets/set.go#L189-L194

Is it enough?

Reply:
We need to sort them in order to get stable diff output, but not for equality

Reply:
I tried with the following

```go
if !names.Equal(testCase.expectedNames) {
				t.Fatal("======")
			}
```

and then

```
go test -count 10000 -run TestPromotedTagsWithRequiredImages ./pkg/steps/release/...
```

It never failed. Isn't the diff stable enough?

Reply:
For my own understand, the sorting is only for pretty output on diff.
We could just fail the test as usual otherwise.